ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/retriever/SimpleSidecarRetriever.java

```diff
@@ -109,34 +107,45 @@ public void flush() {
   public void onNewValidatedSidecar(final DataColumnSidecar sidecar) {
     final DataColumnSlotAndIdentifier dataColumnSlotAndIdentifier =
         DataColumnSlotAndIdentifier.fromDataColumn(sidecar);
-    final List<Map.Entry<DataColumnSlotAndIdentifier, RetrieveRequest>> filteredRequests =
-        pendingRequests.entrySet().stream()
-            .filter(request -> request.getKey().equals(dataColumnSlotAndIdentifier))
-            .filter(request -> !request.getValue().result.isDone())
-            .toList();
-    filteredRequests.forEach(requestEntry -> reqRespCompleted(requestEntry.getValue(), sidecar));
+
+    Optional.ofNullable(pendingRequests.get(dataColumnSlotAndIdentifier))
+        .filter(request -> !request.result.isDone())
+        .ifPresent(request -> reqRespCompleted(request, sidecar));
   }
 
-  private List<RequestMatch> matchRequestsAndPeers() {
-    disposeCompletedRequests();
+  private Stream<RequestMatch> matchRequestsAndPeers() {
     final RequestTracker ongoingRequestsTracker = createFromCurrentPendingRequests();
     return pendingRequests.entrySet().stream()
         .filter(entry -> entry.getValue().activeRpcRequest == null)
         .sorted(Comparator.comparing(entry -> entry.getKey().slot()))
         .flatMap(
             entry -> {
-              RetrieveRequest request = entry.getValue();
+              final RetrieveRequest request = entry.getValue();
```

Comment:
we must have a broken plugin, this shouldnt be able to not be final if our plugin is working